shoop/front/models/stored_basket.py

```diff
@@ -0,0 +1,42 @@
+# -*- coding: utf-8 -*-
+# This file is part of Shoop.
+#
+# Copyright (c) 2012-2015, Shoop Ltd. All rights reserved.
+#
+# This source code is licensed under the AGPLv3 license found in the
+# LICENSE file in the root directory of this source tree.
+
+from django.conf import settings
+from django.db import models
+from picklefield.fields import PickledObjectField
+from shoop.core.fields import MoneyField
+import random
+
+ID_KEYSPACE = "0123456789abcdefghijklmnopqrstuvwxyz"
+
+
+def generate_key():
+    return "".join(random.choice(ID_KEYSPACE) for x in range(32))
+
+
+class StoredBasket(models.Model):
+    # A combination of the PK and key is used to retrieve a basket for session situations.
+    key = models.CharField(max_length=32, default=generate_key)
+    owner_contact = models.ForeignKey("shoop.Contact", blank=True, null=True)
+    owner_user = models.ForeignKey(settings.AUTH_USER_MODEL, blank=True, null=True)
+
+    created_on = models.DateTimeField(auto_now_add=True, db_index=True, editable=False)
+    updated_on = models.DateTimeField(auto_now=True, db_index=True, editable=False)
+    persistent = models.BooleanField(db_index=True, default=False)
+    deleted = models.BooleanField(db_index=True, default=False)
+    finished = models.BooleanField(db_index=True, default=False)
+    title = models.CharField(max_length=64, blank=True)
+    data = PickledObjectField(compress=True)
+
+    # For statistics etc., as `data` is opaque:
+    taxless_total = MoneyField(default=0)
+    taxful_total = MoneyField(default=0)
```

Comment:
IMHO we should just store the total_price and a flag that tells if it's taxful or taxless, since toggling taxfulness of a price can be costly operation (literally, since that needs a request to tax module and e.g. Avalara charges per transaction).